cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueProducer.java

```diff
@@ -29,37 +34,36 @@
 import java.io.Closeable;
 import java.io.IOException;
 import java.util.List;
+import java.util.Set;
 
 /**
  * A {@link co.cask.cdap.data2.queue.QueueProducer} that uses HBase as the storage for queue entries and consumers
- * states. The row key has the following format:
- *
- * <pre>
- * {@code
- *
- * row_key = <salt_prefix> <row_key_base>
- * salt_prefix = 1 byte hash of <row_key_base>
- * row_key_base = <queue_prefix> <write_point> <counter>
- * queue_prefix = <name_hash> <queue_name>
- * name_hash = First byte of MD5 of <queue_name>
- * queue_name = flowlet_name + "/" + output_name
- * write_pointer = 8 bytes long value of the write pointer of the transaction
- * counter = 4 bytes int value of a monotonic increasing number assigned for each entry written in the same transaction
- * }
- * </pre>
+ * states.
  */
 public class HBaseQueueProducer extends AbstractQueueProducer implements Closeable {
 
   private final HBaseQueueStrategy queueStrategy;
+  private final List<ConsumerGroupConfig> consumerGroupConfigs;
   private final byte[] queueRowPrefix;
   private final HTable hTable;
   private final List<byte[]> rollbackKeys;
 
   public HBaseQueueProducer(HTable hTable, QueueName queueName,
-                            QueueMetrics queueMetrics, HBaseQueueStrategy queueStrategy) {
+                            QueueMetrics queueMetrics, HBaseQueueStrategy queueStrategy,
+                            Iterable<? extends ConsumerGroupConfig> consumerGroupConfigs) {
     super(queueMetrics, queueName);
     this.queueStrategy = queueStrategy;
-    // Base row key = queue_name + writePointer + counter
+    // Make sure only one config per consumer group
+    this.consumerGroupConfigs = ImmutableList.copyOf(
+      Iterables.filter(consumerGroupConfigs, new Predicate<ConsumerGroupConfig>() {
+        private final Set<Long> seenGroups = Sets.newHashSet();
+          @Override
+          public boolean apply(ConsumerGroupConfig config) {
+            return seenGroups.add(config.getGroupId());
```

Comment:
This is a placeholder for now and it might change a bit. The original idea was just to have all consumer configs being passed in and only get the unique groups in here (individual consumer config has the same group config if they are in the same group). Since we introduced ConsumerGroupConfig class, it probably better to do it at the queue client factory level. Will have that change when actually hooking up with shared queue